app/kumactl/cmd/config/config_control_planes_add.go

```diff
@@ -76,6 +90,7 @@ func newConfigControlPlanesAddCmd(pctx *kumactl_cmd.RootContext) *cobra.Command
 	cmd.Flags().StringVar(&args.clientKeyFile, "client-key-file", "", "path to the certificate key of a client that is authorized to use the Admin operations of the Control Plane (kumactl stores only a reference to this file)")
 	cmd.Flags().StringVar(&args.caCertFile, "ca-cert-file", "", "path to the certificate authority which will be used to verify the Control Plane certificate (kumactl stores only a reference to this file)")
 	cmd.Flags().BoolVar(&args.skipVerify, "skip-verify", false, "skip CA verification")
+	args.addHeaders = cmd.Flags().StringArray("add-header", nil, "add these headers while communicating to control plane")
```

Comment:
There is `StringToStringVar` take a look at `install control plane`, so you can just receive the map and don't have to parse it yourself

Reply:
(the format then will be `--add-header abc=value`)